joblib/memory.py

```diff
@@ -829,6 +829,8 @@ def __init__(self, location=None, backend='local', cachedir=None,
                               "cachedir value will be ignored.",
                               DeprecationWarning, stacklevel=2)
 
+        self.location = location
+        self.cachedir = location
```

Comment:
I would expect that `cachedir` is only set with the default store backend. Maybe we should use a property for `cachedir` so that you can issue a DeprecationWarning when you access it.

I am not 100% sure whether we should expose `location` at the `Memory` level rather than at the store level (my feeling is probably yes).

Reply:
> Maybe we should use a property for cachedir

Good idea.

> expose location at the Memory level rather than at the store level

if scikit-learn needs an access to the cachedir attribute at the Memory level and since it's one parameter of the Memory constructor, I would say yes. What do you think ?